File: Signiture-Backend/delivery/api/src/main/java/pcy/study/api/common/api/error/TokenErrorCode.java
package pcy.study.api.common.api.error;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import pcy.study.api.common.api.ApiCode;

@Getter
@RequiredArgsConstructor
public enum TokenErrorCode implements ApiCode {

    TOKEN_EXCEPTION(HttpStatus.BAD_REQUEST, 2000, "알 수 없는 토큰 에러입니다."),
    INVALID_TOKEN(HttpStatus.BAD_REQUEST, 2001, "유효하지 않은 토큰입니다."),
    EXPIRED_TOKEN(HttpStatus.BAD_REQUEST, 2002, "만료된 토큰입니다."),
    ;

    private final HttpStatus httpStatus;
    private final Integer code;
    private final String message;
}

File: Signiture-Backend/kotlin/delivery/common/src/main/kotlin/pcy/study/common/api/code/TokenErrorCode.kt
package pcy.study.common.api.code

import pcy.study.common.api.ApiCode

enum class TokenErrorCode(
    private val httpStatusCode: Int,
    private val code: Int,
    private val message: String
) : ApiCode {

    TOKEN_EXCEPTION(400, 2000, "알 수 없는 토큰 에러입니다."),
    INVALID_TOKEN(400, 2001, "유효하지 않은 토큰입니다."),

    EXPIRED_TOKEN(401, 2002, "만료된 토큰입니다."),
    AUTHORIZATION_TOKEN_NOT_FOUND(401, 2003, "인증 정보가 없습니다."),
    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getCode(): Int {
        return this.code
    }

    override fun getMessage(): String {
        return message
    }
}